### PR TITLE
add 160. Intersection of Two Linked Lists

### PR DESCRIPTION
**Pull request by nayuta-ai**

### CODE REVIEW

File: 160_Intersection_of_Two_Linked_Lists.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# class ListNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.next = None
+# time O(m+n) m:length of headA, n:length of headB
+# space O(m+n)
+class Solution:
+    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
+        if not headA or not headB:
+            return None
```
Comment (tkkawa) [inline 160_Intersection_of_Two_Linked_Lists.py:11]: Insert a blank line after the return statement

File: 160_Intersection_of_Two_Linked_Lists.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.next = None
+# time O(m+n) m:length of headA, n:length of headB
+# space O(m+n)
```
Comment (tkkawa) [inline 160_Intersection_of_Two_Linked_Lists.py:7]: time : O( max(m, n) )
space : O(1)

**Review by tkkawa — CHANGES_REQUESTED** [review]

File: 160_Intersection_of_Two_Linked_Lists.py (hunk 2 above)
Comment (nayuta-ai) [reply 160_Intersection_of_Two_Linked_Lists.py]: revised time and space complexity

File: 160_Intersection_of_Two_Linked_Lists.py (hunk 1 above)
Comment (nayuta-ai) [reply 160_Intersection_of_Two_Linked_Lists.py:11]: - I insert a blank line after the return statement

**Review by tkkawa — APPROVED** [review]
LGTM

**Pull request merged**